.azure/validate.azure-pipelines.yml

```diff
@@ -1,28 +1,31 @@
 name: 'Validate the code'
 
 pr:
-  - master
-  - develop
+    - master
+    - develop
 
 pool:
-  vmImage: 'ubuntu-latest'
-  demands:
-  - npm
+    vmImage: 'ubuntu-latest'
+    demands:
+        - npm
 
 steps:
-  - task: Npm@1
-    displayName: 'Install node modules'
-    inputs:
-      verbose: false
+    - task: Npm@1
+      displayName: 'Install node modules'
+      inputs:
+          verbose: false
 
-  - script: npm run types:check
-    displayName: 'Check Types with TypeScript'
+    - script: npm run types:check
+      displayName: 'Check Types with TypeScript'
 
-  - script: npm run format:check
-    displayName: 'Check Format with Prettier'
+    - script: npm run format:check
+      displayName: 'Check Format with Prettier'
 
-  - script: npm run lint
-    displayName: 'Run Linter'
+    - script: npm run lint
+      displayName: 'Run Linter'
 
-  - script: npm run build
-    displayName: 'Build Application'
\ No newline at end of file
+    - script: npm run build
+      displayName: 'Build Application'
+
+    - script: npm run test:coverage
```

Comment:
I was thinking about that, but tests can fail if the build is wrong

Reply:
I also was thinking to move tests to a separated pipeline, I do not have clear vision why yet but for example we will have several types of tests

Reply:
> I was thinking about that, but tests can fail if the build is wrong

I didn't get this point. I think before building we first need to check if there are no errors, and only then build it. Because build may be buildable but it is useless if tests are failing after. So you spend time on building something what you don't know if is valid.

> I also was thinking to move tests to a separated pipeline, I do not have clear vision why yet but for example we will have several types of tests

I don't think it's our case in the nearest future or at all for this kind of (small) project

Reply:
> I didn't get this point. I think before building we first need to check if there are no errors, and only then build it. Because build may be buildable but it is useless if tests are failing after. So you spend time on building something what you don't know if is valid.

Agree about that, but my point is:
Imagine test goes before the build, so there is a chance that the built is broken (incorrect URL or what ever), in that case, the test will fail and the pipeline will show that the test task failed. But it is not a test problem, it was the problem of the build, this is why I wanted to make sure that the build is correct first.

But again, it is a matter of the choice, what do you think about that?

Reply:
I understand your point and I agree. I think it would mostly depend on a project and what takes longer for it: tests or build, if one of them is faster, then this should run first (to save on CI runs time). For our project it doesn't really matter, so it's up to you to change it or leave it.